src/appleseed/renderer/kernel/shading/shadingpoint.h

```diff
@@ -188,6 +188,9 @@ class ShadingPoint
     // Return the material at the intersection point, or 0 if there is none.
     const Material* get_material() const;
 
+    // Return the back material at the intersection point, or 0 if there is none.
+    const Material* get_opposite_material() const;
```

Comment:
I take this back, I now understand that `get_material()` return the front or back material, depending on which side of the surface was hit, and `get_opposite_material()` return the material on the opposite side. The comments for both methods should be updated to clarify this. Maybe:

    // Return the material of the side (front or back) that was hit, at the intersection point, or 0 if there is none.
    const Material* get_material() const;

    // Return the material of the opposite side to the side that was hit, at the intersection point, or 0 if there is none.
    const Material* get_opposite_material() const;